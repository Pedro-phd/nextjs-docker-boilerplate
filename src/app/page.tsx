'use client'

import * as Styled from './page.style'

export default function Home() {
  return (
    <Styled.Main data-testid="home">
      <Styled.Wrapper>
        <Styled.Image data-testid="home-image" src="/next.svg" alt="Next.js Logo" width={180} height={37} priority />
      </Styled.Wrapper>
    </Styled.Main>
  )
}
