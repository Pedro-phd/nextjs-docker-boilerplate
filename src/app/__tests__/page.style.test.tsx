import renderer from 'react-test-renderer'
import { Main, Wrapper, Image } from '../page.style'

describe('Home page styles test', () => {
  it('should render `Main` as expected', () => {
    const tree = renderer.create(<Main />).toJSON()
    expect(tree).toMatchSnapshot('Home - `Main` styled component')
  })

  it('should render `Wrapper` as expected', () => {
    const tree = renderer.create(<Wrapper />).toJSON()
    expect(tree).toMatchSnapshot('Home - `Wrapper` styled component')
  })

  it('should render `Image` as expected', () => {
    const tree = renderer.create(<Image src="" alt="" />).toJSON()
    expect(tree).toMatchSnapshot('Home - `Image` styled component')
  })
})
