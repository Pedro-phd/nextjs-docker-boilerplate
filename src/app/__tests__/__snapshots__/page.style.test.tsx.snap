// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page styles test should render \`Image\` as expected: Home - \`Image\` styled component 1`] = `
<img
  alt=""
  className="sc-hIFLYB"
  data-nimg="1"
  decoding="async"
  onError={[Function]}
  onLoad={[Function]}
  src=""
  style={
    {
      "color": "transparent",
    }
  }
/>
`;

exports[`Home page styles test should render \`Main\` as expected: Home - \`Main\` styled component 1`] = `
<main
  className="sc-eLqIir gKSacg"
/>
`;

exports[`Home page styles test should render \`Wrapper\` as expected: Home - \`Wrapper\` styled component 1`] = `
<div
  className="sc-HzRMQ"
/>
`;
